omod/src/main/java/org/openmrs/module/kenyaemr/web/controller/KenyaemrCoreRestController.java

```diff
@@ -2004,15 +2004,9 @@ else if(pepAndHaartRegimenObsList != null && !pepAndHaartRegimenObsList.isEmpty(
         patientSummary.put("iosResults", iosResults);
 
         //current art regimen
-        CalculationResult currentRegimenResults = EmrCalculationUtils.evaluateForPatient(CurrentArtRegimenCalculation.class, null, patient);
-        if(currentRegimenResults != null) {
-            String roCurrent = currentRegimenResults.toString();
-            if (roCurrent != null) {
-                patientSummary.put("currentArtRegimen", currentRegimenResults.toString());
-            }
-        }
-        else {
-            patientSummary.put("currentArtRegimen", "");
+        Encounter lastEnc = EncounterBasedRegimenUtils.getLastEncounterForCategory(patient, "ARV");
+        if(lastEnc != null) {
+            patientSummary.put("currentArtRegimen", EncounterBasedRegimenUtils.buildRegimenChangeObject(lastEnc.getObs(), lastEnc));
```

Comment:
With this object you can now extract regimen name and startDate. I hope you are not getting regimen start date separately?

Reply:
im getting it from the object